### PR TITLE
Testing/cocktail details

### PR DESCRIPTION
**Pull request by Melizzo**
# PR Description
Please include a summary of the change:
Testing for Cocktail Details Card

Commit Emojis:
:tada: Initial commit
:bar_chart: new tests
:white_check_mark: add or update tests

If it is a new feature, describe the feature and the iteration of the project spec it addresses.
Final iteration - cleaning up test

Note any questions or struggles as well.
Struggle was figuring how to use hooks in testing

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Tests (non-breaking change which adds testing)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [ ] My code follows the Turing style guidelines
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/CocktailDetails/CocktailDetails.test.tsx [hunk 1]
```diff
@@ -1,70 +1,181 @@
...
+          setMadeCocktails={Function}
+        />
+      </MemoryRouter>
+		);
+
+		const favButton = getByText('Mark as Made')
+
+		fireEvent.click(favButton)
+
+		expect(mockToggle).toHaveBeenCalled()
```
Comment (Melizzo) [inline src/CocktailDetails/CocktailDetails.test.tsx:177]: This was tricky to figure how to test if a button was clicked, as the actual setMadeCocktails was handled through a clickHandler, and had to figure out which function to test - ie mockToggle.

File: src/CocktailDetails/CocktailDetails.test.tsx [hunk 2]
```diff
@@ -1,70 +1,181 @@
...
+    const ingredient4 = await waitFor(() => getByText("Ice"));
+    const strMeasure4 = await waitFor(() => getByText("lots"));
+    const ingredient5 = await waitFor(() => getByText("Candy"));
+    const strMeasure5 = await waitFor(() => getByText("1 dash"));
 
-		expect(ingredient1).toBeInTheDocument();
-		expect(ingredient2).toBeInTheDocument();
-		expect(ingredient3).toBeInTheDocument();
+    expect(ingredient1).toBeInTheDocument();
+    expect(strMeasure1.length).toEqual(3);
```
Comment (Melizzo) [inline src/CocktailDetails/CocktailDetails.test.tsx:122]: Thank you Alex for helping me figure out the length testing, cool stuff I'll tuck away for future testing

File: src/CocktailDetails/CocktailDetails.test.tsx [hunk 3]
```diff
@@ -1,70 +1,181 @@
...
+describe("CocktailDetails", () => {
+  mocked(getCocktailDetails).mockImplementation(() =>
+    Promise.resolve({
+      idDrink: "11007",
+      strDrink: "Margarita",
+      strInstructions:
+        "Rub the rim of the glass with the lime slice to make the salt stick to it. Take care to moisten only the outer rim and sprinkle the salt on it. The salt should present to the lips of the imbiber and never mix into the cocktail. Shake the other ingredients with ice, then carefully pour into the glass.",
+      strDrinkThumb:
+        "https://www.thecocktaildb.com/images/media/drink/5noda61589575158.jpg",
+      strIngredient1: "Vodka",
```
Comment (Melizzo) [inline src/CocktailDetails/CocktailDetails.test.tsx:20]: Rubberducking with Becca - super helpful when I didn't mock everything out!!

File: src/CocktailDetails/CocktailDetails.test.tsx (hunk 1 above)
Comment (LeahWinters) [reply src/CocktailDetails/CocktailDetails.test.tsx:177]: Nice job figuring this out! That would have taken me forever.

**Comment by LeahWinters** [comment]
Great job getting so many tests done! Looks really good :)

File: src/CocktailDetails/CocktailDetails.test.tsx [hunk 4]
```diff
@@ -1,70 +1,181 @@
...
+      </MemoryRouter>
+    );
 
-		const ingredient1 = await waitFor(() => getByText('Vodka 1/3 part'));
-		const ingredient2 = await waitFor(() => getByText('Grape soda 1/3 part'));
-		const ingredient3 = await waitFor(() => getByText('Orange juice 1/3 part'));
-		const ingredient4 = await waitFor(() => getByText('Ice lots'));
-		const ingredient5 = await waitFor(() => getByText('Candy 1 dash'));
+    const ingredient1 = await waitFor(() => getByText("Vodka"));
+    const strMeasure1 = await waitFor(() => getAllByText("1/3 part"));
```
Comment (asodemann18) [inline src/CocktailDetails/CocktailDetails.test.tsx:113]: Nice getByAllText!

File: src/CocktailDetails/CocktailDetails.test.tsx [hunk 5]
```diff
@@ -1,70 +1,181 @@
...
+      </MemoryRouter>
+		);
+
+		const favButton = getByText('Mark as Made')
+
+		fireEvent.click(favButton)
+
+		expect(mockToggle).toHaveBeenCalled()
+		expect(getByText("Remove from Made")).toBeInTheDocument()
+	})
```
Comment (st0nebreaker) [inline src/CocktailDetails/CocktailDetails.test.tsx:179]: Wow so many tests. You're a rockstar, Melissa!

File: src/CocktailDetails/CocktailDetails.test.tsx (hunk 1 above)
Comment (asodemann18) [reply src/CocktailDetails/CocktailDetails.test.tsx:177]: nice job!

**Review by asodemann18 — APPROVED** [review]
Looks good!

**Pull request merged**